src/dmd/typesem.d

```diff
@@ -2387,8 +2387,12 @@ Expression getProperty(Type t, Scope* scope_, const ref Loc loc, Identifier iden
                     error(loc, "no property `%s` for type `%s`", ident.toChars(), mt.toPrettyChars(true));
                     if (auto dsym = mt.toDsymbol(scope_))
                         if (auto sym = dsym.isAggregateDeclaration())
+                        {
                             if (auto fd = search_function(sym, Id.opDispatch))
                                 errorSupplemental(loc, "potentially malformed `opDispatch`. Use an explicit instantiation to get a better error message");
+                            else if (!sym.members)
+                                errorSupplemental(loc, "`%s %s` is opaque, therefore it does not have a body.", sym.kind, mt.toPrettyChars(true));
```

Comment:
- `does not have a body` is redundant
- the hint should point to the type declaration, not it's use.

```suggestion
                                errorSupplemental(sym.loc, "`%s %s` is opaque and has no members", sym.kind, mt.toPrettyChars(true));
```